### ISSUE TITLE
Add Stream Capability to the Native Audio Player

### ISSUE THREAD

**Issue by Enverex** [Neos-Metaverse/NeosPublic#2407]
## Is your feature request related to a problem? Please describe.
Currently the video player has a tickbox to stream content rather than downloading it, but there appears to be no way to do the same thing with the native audio player.

## Relevant issues
None.

## Describe the solution you'd like
A tickbox on the StaticAudioClip to set as a stream, or alternatively a different component to handle audio stream URLs which can be fed into the current audio player.

## Describe alternatives you've considered
The video player has "quirks" and doesn't seem to like having MaximumPositionError set (it doesn't work well or perhaps at all, and can easily lead to crashing).

**Comment by Frooxius** [comment]
This isn't matter of just adding a tickbox, the streaming functionality needs to be supported by the actual audio system. What kind of audio streaming mechanisms do you want to use?

This won't likely map to the StaticAudioClip, since that deals specifically with audio clip assets, rather than streams, we'll have to add another source that can handle streaming sources natively.

**Comment by Aerizeon** [comment]
In my usage scenario (and I believe his as well), it would be similar to how the video player supports streaming MP4 files - it would download as needed, instead of all at once.

We're both using it for multi-channel audio (5.1 or 7.1) that has been split out into multiple single-channel files, which can have quite large sizes (50+ mb per channel, except for LFE, which is usually smaller)

Optimally, neos would offer some way of spatializing a 5.1 or 7.1 channel audio stream directly from a video, but I believe you mentioned that having some additional complexities (and is presumably dependent on the new VLC library as well)

I also personally have some really large compilation audio files that end up ~200mb, and it would be nice to have these in-world, without clogging up incoming users' gather jobs with them, and having them streamed as needed instead.

**Comment by Frooxius** [comment]
Most of those features are outside of the scope of this topic and are separate issues. What I'm asking about is what formats and streaming protocols are you looking to use with this?

Streaming audio is a pretty large topic on itself, so having a better understanding of what kind of tech you're using for it is crucial.

**Comment by Aerizeon** [comment]
I'm not sure I understand.

The point of my previous message was to explain the specific use case we have, and the rationale for the request, which is something you generally insist we do. 


 We are talking about being able to play back an incompletely downloaded standard audio file (ogg, wav, flac), similar to how the "stream"  option on the video player functions when given a static video file.
Meaning, if we give the audio player a 300MB OGG  file, it can begin playing shortly after downloading part of the file, instead of requiring the entire 300MB OGG file to complete first.

This was not in reference to SHOUTcast/Icecast streaming, or other audio streaming formats, although I'm sure others would appreciate that.

**Comment by Frooxius** [comment]
Yes, but I also need to know which formats specifically you're interested in having support for, since doing this for native audio playback requires writing custom code for each format, it's not something that can be done generally for all of them. I need to know which ones are the best to cover and what priority they have.

Out of OGG, WAV and FLAC, which one would be top priority for this usecase? 

From what @Enverex mentioned in initial post, sounds more like he's also interested in audio streaming protocols perhaps as well?

**Comment by Enverex** [comment]
In my case I was referring to streaming in a static audio file (e.g. an OGG or AAC/MP4 audio file on a web server), not an actual live stream, for the same reasons @Aerizeon mentioned (being able to play the audio file immediately and slowly stream it, rather than having to download the entire (x)MB file first before it can be played.

Given what would typically be used in this scenario, I'd order them as OGG, FLAC, WAV (I assume AAC/MP4 can't be done as it's not natively supported by Unity?).
